### PR TITLE
Add CLI backend subcommand

### PR DESCRIPTION
**Pull request by jayvdb**
Has some minor backwards incompatibilities in `ms.create_migration`, which could be avoided, however the next release already has a few such as removal of `.state`.

### CODE REVIEW

File: butane_cli/src/lib.rs [hunk 1]
```diff
@@ -231,6 +237,55 @@ pub fn load_connspec(base_dir: &PathBuf) -> Result<db::ConnectionSpec> {
...
+            std::process::exit(1);
+        }
+    } else {
+        // Verify named backends were in the latest migration.
+        if let Ok(ms) = get_migrations(base_dir) {
+            if let Some(latest_migration) = ms.latest() {
+                if let Ok(migration_backends) = latest_migration.sql_backends() {
+                    for backend_name in backend_names {
+                        if !migration_backends.contains(backend_name) {
+                            eprintln!("Adding new backend {backend_name} is not supported.  Use collapse to use a new backend.");
```
Comment (jayvdb) [inline butane_cli/src/lib.rs:271]: "Use collapse to use a new backend." doesnt make sense yet , as collapse doesnt allow `--backend` args.

**Comment by jayvdb** [comment]
CI failure here is the same as https://github.com/diesel-rs/diesel/issues/3960

**Comment by jayvdb** [comment]
CI fixed

File: butane_cli/src/main.rs [hunk 2]
```diff
@@ -20,6 +21,16 @@ struct Cli {
 enum Commands {
     /// Initialize the database.
     Init(InitCommand),
+    /// Add a backend to existing migrations.
+    AddBackend {
```
Comment (jayvdb) [inline butane_cli/src/main.rs:25]: I think I want to change this again, so it is `butane backends [add|remove|list]` reducing this functionality to a single top level sub-command

**Review by Electron100 — APPROVED** [review]

File: butane_cli/src/lib.rs [hunk 3]
```diff
@@ -65,7 +67,9 @@ pub fn init(base_dir: &PathBuf, name: &str, connstr: &str, connect: bool) -> Res
     Ok(())
 }
 
-pub fn make_migration(base_dir: &PathBuf, name: Option<&String>) -> Result<()> {
+/// Make a migration.
+/// When `backend_names` is empty, the backend is selected from the existing migrations or connection.
```
[the remark this answers is not in the document]
Comment (jayvdb) [reply butane_cli/src/lib.rs:71]: It did exist in an earlier version - will update.

**Pull request merged**